### PR TITLE
Add a combination of branch and modified behaviour

### PR DESCRIPTION
**Pull request by lugray**
How do you feel about adding this?

### CODE REVIEW

**Comment by jasoncodes** [comment]
Sounds great. Thanks for the PR! I have wanted this myself on occasion. I may even change my mapping for `CtrlPBranch` to `CtrlPBranchModified`.

Two small issues though:

Firstly, we should have double quotes around the script paths to prevent shell expansion (e.g. in case the path contains spaces).

Secondly, it looks like we’ve got some bash-isms in the new script. We’ll either need to replace these with something `/bin/sh` compatible (I usually use something like `"$(dirname "$0")/branch.sh)"`) or update the shebang to be `/bin/bash`. I’d probably go for removing the bash-isms personally as I don’t think they add much here.

(I find the easiest way to test for this sort of thing on macOS is to `brew install dash` and temporarily change the shebang to `/usr/local/bin/dash`.)

**Comment by lugray** [comment]
That `dash` tip is useful!  I'll have to remember that.  I've got it fixed up now.

Thanks for your plugin, and thanks for your willingness to let me tweak it to my needs! 😃

File: autoload/ctrlp/branch_modified.vim [hunk 1]
```diff
@@ -0,0 +1,35 @@
+
+if exists('g:loaded_ctrlp_branch_modified') && g:loaded_ctrlp_branch_modified
+  finish
+endif
+let g:loaded_ctrlp_branch = 1
```
Comment (jasoncodes) [inline autoload/ctrlp/branch_modified.vim]: I think this should be `g:loaded_ctrlp_branch_modified`.

File: autoload/ctrlp/branch_modified.vim [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+let s:branch_modified_var = {
+\  'init':   'ctrlp#branch_modified#init()',
+\  'exit':   'ctrlp#branch_modified#exit()',
+\  'accept': 'ctrlp#acceptfile',
+\  'lname':  'branch',
+\  'sname':  'branch',
```
Comment (jasoncodes) [inline autoload/ctrlp/branch_modified.vim]: Wondering if we should update the names here too.

**Comment by jasoncodes** [comment]
Those changes look good. Spotted a couple little renaming issues in the `.vim` file. Cheers.

**Comment by lugray** [comment]
🤦‍♀️ At least one of us was paying attention.  Should be fixed.

**Comment by jasoncodes** [comment]
Looks good. Thanks again for the PR! :)

**Pull request merged**